java/client/src/main/java/glide/api/models/commands/scan/ClusterScanCursor.java

```diff
@@ -0,0 +1,95 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.scan;
+
+import glide.api.commands.GenericClusterCommands;
+
+/**
+ * A cursor is used to iterate through data returned by cluster SCAN requests.
+ * @see GenericClusterCommands#scan}.
+ * <p>
+ * This interface is used in two ways:
+ * 1. An {@link #initalCursor()} is passed to {@link GenericClusterCommands#scan} to start a cluster SCAN request.
+ * 2. The result of the {@link GenericClusterCommands#scan} call returns a cursor at <code>index 0</code> of the returned
+ * <code>Object[]</code>. This cursor can be supplied again to a call to {@link GenericClusterCommands#scan}, provided
+ * {@link #isFinished()} returns false.
+ * <p>
+ * Note that cursors returned by {@link GenericClusterCommands#scan} may hold external resources. These resources can
+ * be released by calling {@link #releaseCursorHandle()}. However doing so will disallow the cursor from being used
```

Comment:
By design there isn't. If the user tries to use a dead cursor they get a request error (and there's a test specifically for this, mirroring Python). If they try to use something other than the initial cursor constant, or pass in their own cursor implementation instead of using the one returned by scan, they get an illegal argument exception.